packages/xo-web/src/xo-app/sr/tab-general.js

```diff
@@ -2,24 +2,158 @@ import _ from 'intl'
 import Component from 'base-component'
 import HomeTags from 'home-tags'
 import Icon from 'icon'
-import map from 'lodash/map'
+import PropTypes from 'prop-types'
 import React from 'react'
 import Usage, { UsageElement } from 'usage'
 import { addTag, removeTag, getLicense } from 'xo'
 import { connectStore, formatSize } from 'utils'
 import { Container, Row, Col } from 'grid'
-import { createGetObject } from 'selectors'
-import { renderXoItemFromId } from 'render-xo-item'
+import { createGetObjectsOfType, createSelector } from 'selectors'
+import {
+  differenceBy,
+  find,
+  flatten,
+  forEach,
+  groupBy,
+  keyBy,
+  map,
+  mapValues,
+  sumBy,
+} from 'lodash'
 
-const UsageTooltip = connectStore(() => ({
-  vbd: createGetObject((_, { vdi }) => vdi.$VBDs[0]),
-}))(({ vbd, vdi }) => (
-  <span>
-    {vdi.name_label} − {formatSize(vdi.usage)}
-    {vbd != null && <br />}
-    {vbd != null && renderXoItemFromId(vbd.VM)}
-  </span>
-))
+const styles = { margin: '0.2em', padding: '0.1em' }
+
+@connectStore(() => {
+  const getVbds = createGetObjectsOfType('VBD').pick(
+    createSelector(
+      (_, { group }) => group.vdis || [group],
+      vdis => flatten(map(vdis, _ => _.$VBDs))
+    )
+  )
+
+  const getVms = createGetObjectsOfType('VM').pick(
+    createSelector(
+      getVbds,
+      (vbds, vdis) => map(vbds, vbd => vbd && vbd.VM)
+    )
+  )
+
+  return {
+    vbds: getVbds,
+    vms: getVms,
+  }
+})
+class UsageTooltip extends Component {
+  static propTypes = {
+    group: PropTypes.object.isRequired,
+  }
+
+  _getVmsByVdi = createSelector(
+    () => this.props.vbds,
+    () => this.props.vms,
+    createSelector(
+      () => this.props.group.vdis || [this.props.group],
+      vdis => keyBy(vdis, 'id')
+    ),
+    (vbds, vms, vdis) =>
+      mapValues(vdis, vdi => {
+        const vbd = find(vbds, { VDI: vdi.id })
+        return vbd && vms[vbd.VM]
+      })
+  )
+
+  render() {
+    const {
+      group: {
+        baseCopies,
+        id,
+        name_label: nameLabel,
+        usage,
+        vdis,
+        vdiSnapshots,
+      },
+    } = this.props
+    const vmsByVdi = this._getVmsByVdi()
+    let vm
+    return (
+      <ul style={{ margin: 0, padding: '0.1em' }}>
```

Comment:
IMO you should have a special case when there's only one VDI and nothing else and not put it in a `ul` to make it look cleaner.